### PR TITLE
Improve doc for dropout rng, fix doc display bug

### LINKED ISSUE #1004: Improve documentation for `Dropout` and `rngs` argument in `linen.Module.apply()`
Here is an example of `Dropout` in a model definition:
https://github.com/google/flax/blob/d068512a932da3e05b822790a591bac391aeab36/examples/nlp_seq/models.py#L211

Here is the `apply()`, where `rngs` is passed in
https://github.com/google/flax/blob/d068512a932da3e05b822790a591bac391aeab36/examples/nlp_seq/train.py#L206-L207
However the `rng` is not very clearly explained in `apply()`
https://github.com/google/flax/blob/615f40be774e7ed66fd344e8291ac0d48ebcef7d/flax/linen/module.py#L749
The `rngs` seems to be passed to `flax/core/scope.py`
Here is the code for `Dropout` (linen)
https://github.com/google/flax/blob/9b4807840c5cb26ef5e29028e3558d404aee00a0/flax/linen/stochastic.py#L56-L57
Here is the code for  `make_rng()`
https://github.com/google/flax/blob/615f40be774e7ed66fd344e8291ac0d48ebcef7d/flax/core/scope.py#L441-L447

The documentation for `rngs` in `apply()` should have a (pointer to) list of names of possible rngs
And documentation for `Dropout` should mention how to pass in rng using `apply()`, without directly passing in like `Dropout()(x,rng=rng)`.
Also probably need to mention the `make_rng()` `fold_in` the rng so each dropout layer will use different rng if there are multiple dropout layers.

### PR DESCRIPTION
**Pull request by cccntu**
I am trying to fix #1004
While reading the codebase to know where to edit, I found some display bug.
I will keep pushing to this branch until #1004 is resolved.

### CODE REVIEW

**Comment by cccntu** [comment]
@googlebot I signed it!

File: flax/linen/module.py [hunk 1]
```diff
@@ -809,7 +809,8 @@ def apply(self, variables: VariableDict, *args, rngs: RNGSequences = None,
       variables: A dictionary containing variables keyed by variable
         collections. See :mod:`flax.core.variables` for more details
         about variables.
-      rngs: The rngs for the variable collections.
+      rngs: The rngs for the variable collections. For example :class:`flax.linen.stochastic.Dropout`
```
Comment (8bitmp3) [inline flax/linen/module.py:812]: Small suggestion: I'd recommend using "RNG" instead of "rng" for consistency, since it's an acronym

File: flax/linen/stochastic.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ def __call__(self, inputs, deterministic: Optional[bool] = None, rng=None):
         masked, whereas if true, no mask is applied and the inputs are returned
         as is.
       rng: an optional `jax.random.PRNGKey`. By default `nn.make_rng()` will
-        be used.
+        be used. `nn.make_rng()` requires passing in a rng named `dropout` to `apply()`.
```
Comment (marcvanzee) [inline flax/linen/stochastic.py:51]:
```suggestion
        be used. `nn.make_rng()` requires passing in an RNG named `dropout` to `apply()`.
```

File: flax/linen/module.py [hunk 3]
```diff
@@ -809,7 +809,8 @@ def apply(self, variables: VariableDict, *args, rngs: RNGSequences = None,
       variables: A dictionary containing variables keyed by variable
         collections. See :mod:`flax.core.variables` for more details
         about variables.
-      rngs: The rngs for the variable collections.
+      rngs: The RNGs for the variable collections. For example :class:`flax.linen.stochastic.Dropout`
```
Comment (andsteing) [inline flax/linen/module.py:812]: Nit: can you wrap the line at 80 characters?

File: flax/linen/module.py [hunk 4]
```diff
@@ -809,7 +809,8 @@ def apply(self, variables: VariableDict, *args, rngs: RNGSequences = None,
...
+        requires an additional rng named `dropout`.
```
Comment (andsteing) [inline flax/linen/module.py:813]: Nit: consider using consistent capitalization for "rng".

File: flax/linen/stochastic.py [hunk 5]
```diff
@@ -48,7 +48,7 @@ def __call__(self, inputs, deterministic: Optional[bool] = None, rng=None):
         masked, whereas if true, no mask is applied and the inputs are returned
         as is.
       rng: an optional `jax.random.PRNGKey`. By default `nn.make_rng()` will
-        be used.
+        be used. `nn.make_rng()` requires passing in an RNG named `dropout` to `apply()`.
```
Comment (andsteing) [inline flax/linen/stochastic.py:51]: Nit: could you wrap at 80 characters?

**Review by andsteing — COMMENTED** [review]
Just noticed this PR is assigned to me - sorry for the delay!

**Comment by andsteing** [comment]
Is this PR still alive?

If I don't hear anything within a week, I will close it.

**Comment by cgarciae** [comment]
@andsteing ping me if you are going to close it, I can finish it.

**Comment by andsteing** [comment]
@cgarciae 

closing this PR for now ...
